pennylane/operation.py

```diff
@@ -468,15 +468,27 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
         if wires is None:
             raise ValueError(f"Must specify the wires that {self.name} acts on")
 
-        self._num_params = len(params)
-        # Check if the expected number of parameters coincides with the one received.
-        # This is always true for the default `Operator.num_params` property, but
-        # subclasses may overwrite it to define a fixed expected value.
-        if len(params) != self.num_params:
-            raise ValueError(
-                f"{self.name}: wrong number of parameters. "
-                f"{len(params)} parameters passed, {self.num_params} expected."
-            )
+        # Check whther the expected number of parameters (in case the Operation subclass defines it via the num_params property) coincides with the one received.
+        if hasattr(self.__class__, "num_params"):
+            # pylint: disable=no-member, access-member-before-definition
+            if (
+                isinstance(self.__class__, int)
+                and len(params) != self.__class__.num_params
+                or self.num_params != len(params)
+            ):
+                raise ValueError(
+                    f"{self.name}: wrong number of parameters. "
+                    f"{len(params)} parameters passed, {self.__class__.num_params} specified in class {self.__class__}."
```

Comment:
```suggestion
                    f"{len(params)} parameters passed, {self.__class__.num_params} expected for {self.__class__}."
```

Since this is very user-facing, but users don't need to know what a class specifies.